portal-web/test/functional/com/liferay/portalweb/macros/SAML.macro

```diff
@@ -1,5 +1,138 @@
 definition {
 
+	@description = "Add new IDP Connection data for SP."
+	macro addNewIDPConnection {
+		Navigator.openSpecificURL(url = "${spURL}");
+
+		if (IsElementNotPresent(locator1 = "UserBar#USER_AVATAR_IMAGE")) {
+			if (isSet(spAdminEmailAddress)) {
+				User.loginPG(
+					password = "test",
+					specificURL = "${spURL}",
+					userEmailAddress = "${spAdminEmailAddress}");
+			}
+			else {
+				User.loginPG(
+					password = "test",
+					specificURL = "${spURL}",
+					userEmailAddress = "test@liferay.com");
+			}
+		}
+
+		ApplicationsMenu.gotoPortlet(
+			category = "Security",
+			panel = "Control Panel",
+			portlet = "SAML Admin");
+
+		SAMLNavigation.gotoIdentityProviderConnections();
+
+		SAMLPortlet.addIdentityProviderConnection(
+			idpAttributes = "${idpAttributes}",
+			idpEntityID = "${idpEntityID}",
+			idpKeepAliveURL = "${idpKeepAliveURL}",
+			idpMetadataURL = "${idpMetadataURL}/c/portal/saml/metadata",
+			idpNameID = "${idpEntityID}",
+			nameIdentifierFormat = "${nameIdentifierFormat}");
+	}
+
+	@description = "Add new virtual instance as SP site."
+	macro addNewSite {
+		ApplicationsMenu.gotoPortlet(
+			category = "System",
+			panel = "Control Panel",
+			portlet = "Virtual Instances");
+
+		PortalInstances.addCP(
+			mailDomain = "${mailDomain}",
+			virtualHost = "${virtualHost}",
+			webId = "${webId}");
+	}
```

Comment:
This macro isn't necessary. Also it's not specific to SAML so it shouldn't be in this file.

Reply:
I remove this macro, and put these steps into the test case, thanks!